src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import './GoodsList.scss';
+
+type Props = {
+  goods: string[];
+};
+
+export const GoodsList: React.FC<Props> = ({ goods }) => {
+  const [filter, setFilter] = useState<string>('id');
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const preparedGoods = goods.map((good, index) => ({
+    id: index,
+    name: good,
+  }));
+
+  preparedGoods.sort((g1, g2) => {
+    switch (filter) {
+      case 'id':
+        return (g1.id - g2.id);
+
+      case 'az':
+        return (g1.name.localeCompare(g2.name));
+
+      case 'length':
+        return (g1.name.length - g2.name.length);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
```

Comment:
For now reverse works only single time(then you need to apply sorting using another button), make it reversing the array each time user clicks on it